File: InstantSearchApp/japp/src/main/java/com/woowrale/instantsearch/japp/ui/search/home/SearchPresenter.java
package com.woowrale.instantsearch.japp.ui.search.home;

import com.woowrale.instantsearch.japp.ui.base.BasePresenter;

public class SearchPresenter extends BasePresenter<SearchView> {

    private SearchActivity searchActivity;

    public SearchPresenter(SearchActivity searchActivity) {
        this.searchActivity = searchActivity;
    }

    public void navigateToSearch(String searchType){
        if (searchType.equals("L")){
            searchActivity.navigateToSearchLocal();
        }else{
            searchActivity.navigateToSearchRemote();
        }
    }
}

File: InstantSearchApp/kapp/src/main/java/com/woowrale/instantsearch/kapp/ui/search/home/SearchPresenter.kt
package com.woowrale.instantsearch.kapp.ui.search.home

import com.woowrale.instantsearch.kapp.ui.base.BasePresenter

class SearchPresenter(private val searchActivity: SearchActivity) : BasePresenter<SearchView>() {

    fun navigateToSearch(searchType: String) {
        if (searchType == "L") {
            searchActivity.navigateToSearchLocal()
        } else {
            searchActivity.navigateToSearchRemote()
        }
    }
}